user-console/source/org/pentaho/mantle/client/ui/tabs/MantleTabPanel.java

```diff
@@ -818,7 +820,44 @@ public void onResponseReceived( Request request, Response response ) {
           if ( response.getStatusCode() == Response.SC_OK ) {
             JSONObject jsonObject = (JSONObject) JSONParser.parseLenient( response.getText() );
             if ( jsonObject != null ) {
-              isAsyncModeEnabled = jsonObject.get( "supportAsync" ).isBoolean().booleanValue();
+              isReportViewerAsyncModeEnabled = jsonObject.get( "supportAsync" ).isBoolean().booleanValue();
+            }
+          }
+        }
+      } );
+    } catch ( RequestException e ) {
+      // showError(e);
+    }
+  }
+
+  /**
+   * Queries the server once to see whether PIR async-mode is enabled. If it is enabled,
+   * the report-viewer provides it own loading screen and the default "please wait" screen interferes
+   * with the user experience of that. Therefore we disable the wait-screen that comes up when a new
+   * page opens. @see case BACKLOG-9340 for details.
+   */
+  public void fetchAsyncModeEnabledStatusForPIRPlugin() {
```

Comment:
This code is almost identical to the other method. Refactor so that the changing bit (the static text appended to the URL) is provided as parameter. This will greatly reduce the amount of duplicate code you have in this class.